hermetic_build/library_generation/cli/entry_point.py

```diff
@@ -115,103 +101,63 @@ def generate(
-            "library generation."
+            f"Generation config {generation_config_path} does not exist."
         )
-
-    current_generation_config_path = os.path.abspath(current_generation_config_path)
     repository_path = os.path.abspath(repository_path)
     api_definitions_path = os.path.abspath(api_definitions_path)
-    include_library_names = _parse_library_name_from(library_names)
-
-    if not baseline_generation_config_path:
-        # Execute selective generation based on current_generation_config if
-        # baseline_generation_config is not specified.
-        generate_from_yaml(
-            config=from_yaml(current_generation_config_path),
-            repository_path=repository_path,
-            api_definitions_path=api_definitions_path,
-            target_library_names=include_library_names,
-        )
-        return
-
-    # Compare two generation configs to get changed libraries.
-    baseline_generation_config_path = os.path.abspath(baseline_generation_config_path)
-    config_change = compare_config(
-        baseline_config=from_yaml(baseline_generation_config_path),
-        current_config=from_yaml(current_generation_config_path),
-    )
-    # Pass None if we want to fully generate the repository.
-    changed_library_names = (
-        config_change.get_changed_libraries()
-        if not _needs_full_repo_generation(config_change=config_change)
-        else None
-    )
-    # Include library names takes preference if specified.
-    target_library_names = (
-        include_library_names
-        if include_library_names is not None
-        else changed_library_names
+    generation_config = from_yaml(generation_config_path)
+    include_library_names = _parse_library_name_from(
+        includes=library_names, generation_config=generation_config
     )
     generate_from_yaml(
-        config=config_change.current_config,
+        config=generation_config,
         repository_path=repository_path,
         api_definitions_path=api_definitions_path,
-        target_library_names=target_library_names,
+        target_library_names=include_library_names,
     )
 
 
-def _needs_full_repo_generation(config_change: ConfigChange) -> bool:
+def _needs_full_repo_generation(generation_config: GenerationConfig) -> bool:
     """
     Whether you should need a full repo generation, i.e., generate all
     libraries in the generation configuration.
     """
-    current_config = config_change.current_config
-    return not current_config.is_monorepo() or current_config.contains_common_protos()
+    return (
+        not generation_config.is_monorepo()
+        or generation_config.contains_common_protos()
+    )
 
 
-def _parse_library_name_from(includes: str) -> Optional[list[str]]:
-    if includes is None:
+def _parse_library_name_from(
+    includes: str, generation_config: GenerationConfig
+) -> Optional[list[str]]:
+    if includes is None or _needs_full_repo_generation(generation_config):
```

Comment:
Is this `_needs_full_repo_generation(generation_config)` a logic change?

Reply:
No, this is an existing logic.

Reply:
This function determines whether we want a full generation, i.e., we want a full generation if it's a split repo or sdk-platform-java.

The logic is there before but I moved it here to help create the library name list. If the user doesn't specify `--library-names` or we do want a full generation, returns `None`, which means full generation.